index.bs

```diff
@@ -2581,6 +2665,62 @@ an optional [=boolean=] |isNoised| (default false):
 1. Run [=obtain and deliver a verbose debug report=] with « |data| », |source|'s [=attribution source/reporting origin=],
     and |source|'s [=attribution source/fenced=].
 
+To <dfn>find sources with common destinations and reporting origin</dfn> given an [=attribution source=] |pendingSource|:
+1. Let |matchingSources| be a new [=list=].
+1. [=set/iterate|For each=] |source| of the user agent's [=attribution source cache=]:
+    1. Let |commonDestinations| be the [=set/intersection=] of |source|'s [=attribution source/attribution destinations=] and |pendingSource|'s [=attribution source/attribution destinations=].
+    1. If |commonDestinations| [=set/is empty=], [=iteration/continue=].
+    1. If |source|'s [=attribution source/reporting origin=] and |pendingSource|'s [=attribution source/reporting origin=] are not [=same origin=], [=iteration/continue=].
+    1. [=list/Append=] |source| to |matchingSources|.
+1. Return |matchingSources|.
+
+To <dfn>remove fake event reports for source</dfn> given an [=attribution source=] |source| and a [=moment=] |minTriggerTime|:
+1. If |source|'s [=attribution source/randomized response=] is null, return.
+1. [=set/Remove=] all elements from the [=event-level report cache=] whose [=event-level report/trigger time=] is greater than or equal to |minTriggerTime| and whose [=event-level report/source identifier=] is equal to |source|'s [=attribution source/source identifier=].
+
+To <dfn>remove sources with different max event states if necessary</dfn> given an [=attribution source=] |pendingSource|:
```

Comment:
Why does the language here qualify with "if necessary?" Why not just "remove sources with different max event states?"